harness/determined/cli/experiment.py

```diff
@@ -304,66 +320,99 @@ def describe(args: Namespace) -> None:
-                    checkpoint_state = None
+                    checkpoint_state = ""
                     checkpoint_end_time = None
 
-                validation = step.get("validation")
-                if validation:
-                    validation_state = validation["state"]
-                    validation_end_time = validation.get("end_time")
+                v_metrics_fields = []
+                if step.validation:
+                    step_detail = step.validation
+                    validation_state = step_detail.state.value
+                    validation_end_time = step_detail.endTime
+                    for name in v_metrics_names:
+                        if step_detail.metrics and (name in step_detail.metrics):
+                            v_metrics_fields.append(step_detail.metrics[name])
+                        else:
+                            v_metrics_fields.append(None)
                 else:
-                    validation_state = None
+                    validation_state = ""
                     validation_end_time = None
-
-                if args.metrics:
-                    v_metrics_fields = [
-                        api.metric.get_validation_metric(name, validation)
-                        for name in v_metrics_names
-                    ]
-                else:
-                    v_metrics_fields = []
-
-                row = (
-                    [
-                        step["trial_id"],
-                        step["total_batches"],
-                        step["state"],
-                        render.format_time(step.get("end_time")),
-                    ]
-                    + t_metrics_fields
-                    + [
-                        checkpoint_state,
-                        render.format_time(checkpoint_end_time),
-                        validation_state,
-                        render.format_time(validation_end_time),
-                    ]
-                    + v_metrics_fields
-                )
-                values.append(row)
+                    v_metrics_fields = [None for name in v_metrics_names]
+
+                if step_detail:
+                    if step_detail.totalBatches in step_output:
+                        # condense training, checkpoints, validation workloads into one 'step' row
+                        # for compatibility with previous versions of describe
+                        merge_row = step_output[step_detail.totalBatches]
+                        merge_row[3] = max(merge_row[3], render.format_time(step_detail.endTime))
+                        for idx, tfield in enumerate(t_metrics_fields):
+                            if tfield and merge_row[4 + idx] is None:
+                                merge_row[4 + idx] = tfield
+                        start_checkpoint = 4 + len(t_metrics_fields)
+                        if checkpoint_state:
+                            merge_row[start_checkpoint] = checkpoint_state.replace("STATE_", "")
```

Comment:
if the state is coming from our generated code as an enum you can do `.value` to get it without the prefix IIRC

Reply:
In the bindings, the state object's value returns a string but it still has the STATE_ prefix.

I've tried a few ways of doing this where we store the `determinedexperimentv1State` instead of a string, then handle `.value` and `STATE_` in the output, but it runs into two main problems:
- Some rows have no checkpoint or validation state. We want to keep these blank and not use `STATE_UNSPECIFIED`
- These columns are passed to `render.tabulate_or_csv`, which is expecting strings and will render a state object as `determinedexperimentv1State.STATE_COMPLETED`